### ISSUE TITLE
Bug when cropping images with a higher resolution than the initial image

### ISSUE THREAD

**Issue by roykoper** [sulu/sulu#5385]
| Q | A
| --- | ---
| Sulu Version | 2.1.0
| Browser Version | Google Chrome 81.0.4044.138

# Actual Behavior

Here's a video to demonstrate how it's currently behaving

![SuluCropBug](https://user-images.githubusercontent.com/39400442/83614437-7c211e00-a585-11ea-96dd-d07216009cec.gif)

There's 2 issues here:
1. When changing the format, the resize window does not change until after trying to drag it.
2. It's not possible to drag the resize window to the right

# Expected Behavior

1. The resize window should change after selecting a format, not after dragging the resize window.
2. To be able to drag the resize window to the right

# Steps to Reproduce

1. Install a new Sulu project following the Getting Started(https://docs.sulu.io/en/2.0/book/getting-started.html)

2. Upload an image

3. Create a new image format with a higher width/height than the initial image

```
    <format key="format">
        <meta>
            <title lang="nl, en">(420x720px)</title>
        </meta>
        <scale x="420" y="720"/>

        <options>
            <option name="jpeg_quality">80</option>
            <option name="png_compression_level">6</option>
        </options>
    </format>
```

4. Choose the image format when cropping
5. Try to crop the image
6. Observe

# Possible Solutions

In the demonstration video I've used a 500x500px image, when selecting an image with a higher width/height the issue occurs, but when using a smaller width/height than the initial image it is working as expected.

Previous issue as I'm unable to reopen it, link to it: https://github.com/sulu/sulu/issues/5318

**Comment by alexander-schranz** [comment]
@roykoper Thank you for report. 

1. I can reproduce it in the case for the first selected image format when opening the crop window is not correct.
2. Can also reproduce this, seems to be similar issue as described in #5372

**Comment by roykoper** [comment]
I would be interested in a fix for this, is there an estimation of when this will get picked up?

**Comment by nnatter** [comment]
Hey, 
can you check if #5399 fixes your problem?

**Comment by roykoper** [comment]
Hey, this definitely seems the right way forward! However for certain image sizes I'm still not able to drag it fully to the right as can be seen in the following picture:
![image](https://user-images.githubusercontent.com/39400442/87168425-c1e0ad00-c2ce-11ea-8e7f-c06b28850f14.png)

**Comment by nnatter** [comment]
I suspect that this is an issue with rounding the scaled coordinates / positioning the rectangle on the scaled image, not sure if we can solve this for all cases. Do things behave differently if you change the size/zoom of your browser window? Can you provide your test image and the respective image format?

**Comment by roykoper** [comment]
It does seem to behave differently based on the size of my browser window, my browser size is 1440x725. Increasing/decreasing the size of the browser window does seem to fix it

![image](https://user-images.githubusercontent.com/39400442/87655684-d999b480-c758-11ea-8e24-d595d5fb5d06.png)
This is the image I used, which is 500x500

One of the image formats I'm using that can reproduce it

```
<format key="420x730">
        <meta>
            <title lang="nl, en">Format (420x730px)</title>
        </meta>
        <scale x="420" y="730"/>

        <options>
            <option name="jpeg_quality">80</option>
            <option name="png_compression_level">6</option>
        </options>
</format>
```
The way to reproduce it is to choose an image format with a smaller width and a higher height than the initial image.

**Comment by nnatter** [comment]
Managed to reproduce the behaviour with the exact parameters posted in https://github.com/sulu/sulu/issues/5385#issuecomment-659306337

I am quite sure that this is a minor display issue that results from rounding the scaled coordinates that are used to position the `SelectionRectangle`. The dark border at the right side side (see https://github.com/sulu/sulu/issues/5385#issuecomment-656720950) is visible when the browser window is exactly `1439x725`. It disappears if the browser window is resized to `1438x725` or `1440x725`. In all of these cases, the value that is sent to the server is exactly the same (`{cropWidth: 287.6712328767123, cropHeight: 500, cropX: 212.3287671232877, cropY: 0}`).

I am sure that its possible to find a rounding/positioning strategy that would fix the issue for this specific parameters. But that will likely lead to a similar behaviour for another set of parameters (image size, screen size, image-format ratio) in my opinion.

**Comment by nnatter** [comment]
Managed to reproduce the behaviour with the exact parameters posted in https://github.com/sulu/sulu/issues/5385#issuecomment-659306337

I am quite sure that this is a minor display issue that results from rounding the scaled coordinates that are used to position the `SelectionRectangle`. The dark border at the right side side (see https://github.com/sulu/sulu/issues/5385#issuecomment-656720950) is visible when the browser window is exactly `1439x725`. It disappears if the browser window is resized to `1438x725` or `1440x725`. In all of these cases, the value that is sent to the server is exactly the same (`{cropWidth: 287.6712328767123, cropHeight: 500, cropX: 212.3287671232877, cropY: 0}`).

I am sure that its possible to find a rounding/positioning strategy that would fix the issue for this specific parameters. But that will likely lead to a similar behaviour for another set of parameters (image size, screen size, image-format ratio) in my opinion. Because of this, I do not plan to spend more time on this.

**Comment by roykoper** [comment]
Interesting that the issue only occurs for that specific browser window size. I'm happy that it works for the most part, definitely an improvement! I'm sure I'll be able to work around the one pixel on the right for now. Thank you for looking into this!

**Comment by danrot** [comment]
Fixed in #5399
